influxdata/influxdb_iox
# feat: schema client and CLI

Closes #4055 

Following on from #4081, this adds a client and CLI command to fetch the schema for a namespace.

To test against IOx all-in-one mode:
- launch a postgres (i use a docker container)
- run sqlx setup against it (see catalog docs)
- run psql to put some test data in:

```
psql $ set search_path to iox_catalog, public;
psql $ insert into kafka_topic (name) values ('franz');
psql $ insert into query_pool (name) values ('swimming');
psql $ insert into namespace (name, retention_duration, kafka_topic_id, query_pool_id) values ('namespace_table_test', 'inf', 1, 1);
psql $ insert into table_name (namespace_id, name) values (1, 'schema_test_table');
psql $ insert into column_name (table_id, name, column_type) values (1, 'schema_test_column', 1);
```

- run IOx in all-in-one mode (adjust dsn where appropriate):

```
$ LOG_FILTER=debug cargo run -- run   all-in-one   --write-buffer=file  --write-buffer-addr /tmp  --catalog-dsn postgresql://postgres:password@localhost:5432/iox_shared  --write-buffer-partition-range-start 0  --write-buffer-partition-range-end 0  --pause-ingest-size-bytes 10000  --persist-memory-threshold-bytes 1000
```

- run CLI:

```
$ cargo run -- -h http://localhost:8081 schema get namespace_table_test
    Finished dev [unoptimized + debuginfo] target(s) in 0.17s
     Running `target/debug/influxdb_iox -h 'http://localhost:8081' schema get namespace_table_test`
{
  "id": 1,
  "kafkaTopicId": 1,
  "queryPoolId": 1,
  "tables": {
    "schema_test_table": {
      "id": 1,
      "columns": {
        "schema_test_column": {
          "id": 2,
          "columnType": 1
        }
      }
    }
  }
}
```
